dotCMS/src/test/java/com/liferay/util/FileUtilTest.java

```diff
@@ -2,19 +2,118 @@
+        CloseUtils.closeQuietly(in);
+
+        Assert.assertEquals("The file recovery should be the file saved", new String(buff), string);
+    }
+
+    @Test
+    public void test_compress_gzip_outputstream()  throws Exception {
+
+        final File file = com.dotmarketing.util.FileUtil.createTemporalFile("gzip");
+        final OutputStream out = FileUtil.createOutputStream(file, "gzip");
+
+        final String  string = "testcompress";
+        final byte [] bytes  = string.getBytes();
+        out.write(bytes);
+        CloseUtils.closeQuietly(out);
+
+        final InputStream in   = FileUtil.createInputStream(file, "gzip");
+        final byte [] buff     = new byte[bytes.length];
+        in.read(buff);
+        CloseUtils.closeQuietly(in);
+
+        Assert.assertEquals("The file recovery should be the file saved", new String(buff), string);
+    }
+
+    @Test
+    public void test_compress_gzip_enum_outputstream()  throws Exception {
+
+        final File file = com.dotmarketing.util.FileUtil.createTemporalFile("egzip");
+        final OutputStream out = FileUtil.createOutputStream(file.toPath(), FileUtil.StreamCompressorType.GZIP);
+
+        final String  string = "testcompress";
+        final byte [] bytes  = string.getBytes();
+        out.write(bytes);
+        CloseUtils.closeQuietly(out);
+
+        final InputStream in   = FileUtil.createInputStream(file.toPath(), FileUtil.StreamCompressorType.GZIP);
+        final byte [] buff     = new byte[bytes.length];
+        in.read(buff);
+        CloseUtils.closeQuietly(in);
+
+        Assert.assertEquals("The file recovery should be the file saved", new String(buff), string);
+    }
+
+    @Test
+    public void test_compress_bzip2_outputstream()  throws Exception {
+
+        final File file = com.dotmarketing.util.FileUtil.createTemporalFile("bzip2");
+        final OutputStream out = FileUtil.createOutputStream(file, "bzip2");
+
+        final String  string = "testcompress";
+        final byte [] bytes  = string.getBytes();
+        out.write(bytes);
+        CloseUtils.closeQuietly(out);
+
+        final InputStream in   = FileUtil.createInputStream(file, "bzip2");
+        final byte [] buff     = new byte[bytes.length];
+        in.read(buff);
+        CloseUtils.closeQuietly(in);
+
+        Assert.assertEquals("The file recovery should be the file saved", new String(buff), string);
+    }
+
+    @Test
+    public void test_compress_bzip2_enum_outputstream()  throws Exception {
+
+        final File file = com.dotmarketing.util.FileUtil.createTemporalFile("ebzip2");
+        final OutputStream out = FileUtil.createOutputStream(file.toPath(), FileUtil.StreamCompressorType.BZIP2);
+
+        final String  string = "testcompress";
+        final byte [] bytes  = string.getBytes();
+        out.write(bytes);
+        CloseUtils.closeQuietly(out);
+
+        final InputStream in   = FileUtil.createInputStream(file.toPath(), FileUtil.StreamCompressorType.BZIP2);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)